bin/tomcat-ctl

```diff
@@ -1,199 +1,246 @@
+        curl -L "$WAR_SOURCE_URL" -o $WAR_FILE
+        WAR_URL_FOR_PRINT=`curl -Is $WAR_SOURCE_URL | awk '/Location/ { print $2 }'`
+        log "WAR file downloaded: $WAR_URL_FOR_PRINT"
+        echo_time "WAR file downloaded: $WAR_URL_FOR_PRINT"
+        # Extract the WAR file
+        cd $TOMCAT_KFS_DIR
+        jar -xf $WAR_FILE
+    fi
+
+    log "Starting Tomcat application server tomcat ..."
+    echo_time "Starting Tomcat application server tomcat ..."
+
+    # copy in tomcat configuration files
+    log "Copying tomcat configuration files."
+    echo_time "Copying tomcat configuration files."
+    cp $TOMCAT_CONFIG_DIRECTORY/server.xml $TOMCAT_BASE_DIR/conf/
+    cp $SECURITY_DIRECTORY/tomcat-users.xml $TOMCAT_BASE_DIR/conf/
+    cp $TOMCAT_CONFIG_DIRECTORY/logging.properties $TOMCAT_BASE_DIR/conf
+    cp $TOMCAT_CONFIG_DIRECTORY/log4j.properties $TOMCAT_KFS_WEBINF_DIR/classes
+
+    # copy in New Relic config file
+    log "Copying New Relic configuration file."
+    echo_time "Copying New Relic configuration file."
+    cp $SECURITY_DIRECTORY/newrelic.yml $TOMCAT_SHARE_LIB/
+
+    # copy in oracle JDBC drivers
+    log "Copying oracle JDBC drivers."
+    echo_time "Copying oracle JDBC drivers."
+    cp $TOMCAT_CONFIG_DIRECTORY/classes/* $TOMCAT_SHARE_LIB/
+
+    # Get changelogs for liquibase
+    log "Loading changelog files for liquibase."
+    echo_time "Loading changelog files for liquibase"
+
+    # create new directory to hold the files extracted from kfs-core-ua jar file
+    mkdir -p $TOMCAT_KFS_CORE_DIR
+
+    # create new directory to hold the UA changelog files
+    mkdir -p $UA_DB_CHANGELOGS_DIR
+
+    # extract files in kfs-core-ua* jar
+    cd $TOMCAT_KFS_CORE_DIR
+    /usr/bin/jar -xf $TOMCAT_KFS_DIR/WEB-INF/lib/kfs-core-ua*
+
+    # copy edu changelog files to changelogs directory
+    mv $TOMCAT_KFS_CORE_DIR/edu $UA_DB_CHANGELOGS_DIR
+
+	# run liquibase here
+    log "Running liquibase update."
+    echo_time "Running liquibase update."
+	liquibase_update_kfs.sh $APP_VERSION
+
+    log "Completed running liquibase update"
+    echo_time "Completed running liquibase update"
+
+    cp $TOMCAT_CONFIG_DIRECTORY/setenv.sh $TOMCAT_SHARE_BIN/
+    chmod +x $TOMCAT_SHARE_BIN/setenv.sh
+
+    #run logrotate on our log files
+    logrotate -f /etc/logrotate.conf
+	
+	#if rhubarb-config has not been deployed, deploy it
+    log "Checking for existing rhubarb-config deployment..."
+    echo_time "Checking for existing rhubarb-config deployment..."
+    if [ ! -d /etc/opt/kuali/rhubarb ]
+    then
+        log "No rhubarb-config deployment found, deploying rhubarb"
+        echo_time "No rhubarb-config deployment found, deploying rhubarb"
+        startdir=$(pwd)
+        service ssh start
+        mkdir -p /git-clones
+        git clone "https://github.com/ua-eas/katt-kfs-rhubarb-config" /git-clones/katt-kfs-rhubarb-config
+        cd /git-clones/katt-kfs-rhubarb-config/ && bundle install

```

Comment:
Current default would be 'development'. There's already a KFS 'environment type' variable ($KFS_ENV_NAME) that is passed in as a parameter at runtime, so this section could be enhanced to respect that.

Alternatively, as we discussed earlier w/ Mark, if we move from deploying at runtime to deploying at buildtime, then those different environments will become different containers entirely and tagged accordingly.

I'd rather open a separate ticket to do the work once based on the decision to deployment at runtime vs buildtime.